includes/foundation-events/class-foundation-events.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * Foundation events remote data blocks
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+class Foundation_Events {
+
+	public static function init() {
+		add_action( 'init', [ self::class, 'register_data_blocks' ] );
+
+		add_filter( 'query_vars', [ self::class, 'add_query_vars' ] );
+		add_filter( 'remote_data_blocks_query_input_variables', [ self::class, 'add_query_vars_to_input_variables' ], 10, 2 );
+	}
+
+	public static function register_data_blocks() {
+		if ( ! function_exists( 'register_remote_data_block' ) ) {
+			return;
+		}
+
+		register_remote_data_block(
+			[
+				'title'             => 'Single Foundation Event',
+				'render_query'      => [
+					'query' => self::get_single_event_query(),
+				],
+				'patterns'          => [
+					[
+						'title' => 'Single event 1',
+						'html'  => file_get_contents( __DIR__ . '/pattern-1.html' ),
+					],
+				],
+				'selection_queries' => [
+					[
+						'display_name' => 'Select an event',
+						'query'        => self::get_list_query(),
+						'type'         => 'list',
+					],
+
+				],
+				'overrides'         => [
+					[
+						'name'         => 'event_id_override',
+						'display_name' => __( 'Use event ID from URL', 'newspack' ),
+						'help_text'    => __( 'For use on the /event/ page', 'newspack' ),
+					],
+				],
+			]
+		);
+
+		register_remote_data_block(
+			[
+				'title'        => 'Foundation Events List',
+				'render_query' => [
+					'query' => self::get_list_query(),
+				],
+				'patterns'     => [
+					[
+						'title' => 'Foundation Events List',
+						'html'  => file_get_contents( __DIR__ . '/pattern-2.html' ),
+					],
+				],
+			]
+		);
+
+		add_rewrite_rule( '^event/([0-9]+)/?', 'index.php?pagename=foundation-event&event_id=$matches[1]', 'top' );
+	}
+
+	public static function add_query_vars( $query_vars ) {
+		$query_vars[] = 'event_id';
+		return $query_vars;
+	}
+
+	public static function add_query_vars_to_input_variables( array $input_variables, array $enabled_overrides ): array {
+		if ( true === in_array( 'event_id_override', $enabled_overrides, true ) ) {
+			$event_id = get_query_var( 'event_id' );
+
+			if ( ! empty( $event_id ) ) {
+				$input_variables['event_id'] = $event_id;
+			}
+		}
+
+		return $input_variables;
+	}
+
+	public static function get_data_source() {
+		return [
+			'display_name'    => 'Foundation Events',
+			'endpoint'        => NEWSPACK_FOUNDATION_TEST_EVENTS_API_BASE_URL,
```

Comment:
Note: The site throws fatal errors if this constant is missing. It's unclear what exactly it should be, but I can work it out :)